ap-java-sdk/src/test/java/com/alternativepayments/models/CustomerModelTest.java

```diff
@@ -0,0 +1,145 @@
+        assertThat(customer.getPhone()).isEqualTo("55555555555");
+        assertThat(customer.getMode()).isEqualTo("Live");
+        assertThat(customer.getCreated().toString()).isEqualTo("2012-04-23T18:25:43.511Z");
+    }
+
+    @Test
+    public void return_all_customers() {
+        CustomerApiMock.expectGetAll();
+
+        final CustomerCollection customers = alternativePaymentClient.getAllCustomers();
+
+        assertThat(customers.getCustomers()).hasSize(2);
+        assertThat(customers.getPagination().getOffset()).isEqualTo(10);
+        assertThat(customers.getPagination().getLimit()).isEqualTo(3);
+        assertThat(customers.getPagination().getCount()).isEqualTo(2504);
+
+        final Customer firstCustomer = customers.getCustomers().get(0);
+        assertThat(firstCustomer.getId()).isEqualTo("cus_7f0724f3b1d745d49");
+        assertThat(firstCustomer.getFirstName()).isEqualTo("Jane");
+        assertThat(firstCustomer.getLastName()).isEqualTo("Doe");
+        assertThat(firstCustomer.getEmail()).isEqualTo("jane@doe.com");
+        assertThat(firstCustomer.getAddress()).isEqualTo(null);
+        assertThat(firstCustomer.getAddress2()).isEqualTo(null);
+        assertThat(firstCustomer.getCity()).isEqualTo(null);
+        assertThat(firstCustomer.getZip()).isEqualTo(null);
+        assertThat(firstCustomer.getCountry()).isEqualTo("DE");
+        assertThat(firstCustomer.getState()).isEqualTo(null);
+        assertThat(firstCustomer.getPhone()).isEqualTo(null);
+        assertThat(firstCustomer.getMode()).isEqualTo("Live");
+        assertThat(firstCustomer.getCreated().toString()).isEqualTo("2012-04-23T18:25:43.511Z");
+
+        final Customer secondCustomer = customers.getCustomers().get(1);
+        assertThat(secondCustomer.getId()).isEqualTo("cus_7f0724f3b1d745d49");
+        assertThat(secondCustomer.getFirstName()).isEqualTo("John");
+        assertThat(secondCustomer.getLastName()).isEqualTo("Doe");
+        assertThat(secondCustomer.getEmail()).isEqualTo("john@doe.com");
+        assertThat(secondCustomer.getAddress()).isEqualTo("Rutledge Ave 409");
+        assertThat(secondCustomer.getAddress2()).isEqualTo(null);
+        assertThat(secondCustomer.getCity()).isEqualTo("Folsom");
+        assertThat(secondCustomer.getZip()).isEqualTo("19033");
+        assertThat(secondCustomer.getCountry()).isEqualTo("US");
+        assertThat(secondCustomer.getState()).isEqualTo("PA");
+        assertThat(secondCustomer.getPhone()).isEqualTo("55555555555");
+        assertThat(secondCustomer.getMode()).isEqualTo("Live");
+        assertThat(secondCustomer.getCreated().toString()).isEqualTo("2012-04-23T18:25:43.511Z");
+
+    }
+
+    @Test
+    public void create_new_customer() {
+        Customer customer = new Customer("cus_bd838e3611d34d598", null, null, null, "John", "Doe", "john@doe.com",
+                "Rutledge Ave 409", null, "Folsom", "19033", "US", "PA", "55555555555", null);
+
```

Comment:
Lets create a trello ticket to think of nicer way of creating objects (builder or at least couple of constructors) since this can get ugly with this many properties.